src/eos-metrics-instrumentation.c

```diff
@@ -371,43 +387,33 @@ record_login (GDBusProxy *dbus_proxy,
               GVariant   *parameters,
               gpointer    user_data)
 {
-    if (strcmp ("SessionRemoved", signal_name) == 0 &&
-        remove_session_from_set (parameters))
+    if (strcmp ("SessionRemoved", signal_name) == 0)
       {
-        GVariant *session_id = g_variant_get_child_value (parameters, 0);
-        emtr_event_recorder_record_stop_sync (emtr_event_recorder_get_default (),
-                                              USER_IS_LOGGED_IN,
-                                              session_id,
-                                              NULL /* auxiliary_payload */);
-        g_variant_unref (session_id);
+        const gchar *session_id;
```

Comment:
Yeah, GVariant has a lot of weird features stuffed in it. I noticed you weren't writing very idiomatic GVariant with `get_child_value` and such, so I figured I'd take the time to clean it up.